src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -193,6 +193,8 @@ AndroidDeviceControllerWrapper * AndroidDeviceControllerWrapper::AllocateNew(
     }
     MutableByteSpan rcacSpan(rcac.Get(), kMaxCHIPDERCertLength);
 
+    Crypto::P256Keypair ephemeralKey;
+
```

Comment:
Can you describe why changing the scope here fixes things? As far as I can tell, nothing in the first `if {..}` utilizes `ephemeralKey`?

Reply:
`setupParams` stores a pointer to the key, and is used after the else.  So this is not about the `if`; it's about the code that ran after `ephemeralKey` went out of scope but was using a pointer to it.

The real right fix here is that `setupParams` should be declared after all the things it's going to store pointers to, including `ephemeralKey`....

Reply:
@andy31415 I added comment. Thank you for your comment.
@mrjerryjohns Someone else has already mentioned my point.
@bzbarsky-apple I thought about a better way, but couldn't find a better way, so I modified it like this.